x-pack/plugin/esql/qa/testFixtures/src/main/resources/dissect.csv-spec

```diff
@@ -175,6 +175,14 @@ Parto Bamford     | Parto          | Bamford
 ;
 
 
+overwriteInputName
+from employees | sort emp_no asc | dissect first_name "%{first_name}o%{rest}" | keep emp_no, first_name, rest | limit 1;
```

Comment:
Weird. There's a test already for this case, `shadowingSelf`. Maybe we should add a unit test for the analyzer to test explicitly what's going on?

At least, let's add a comment that says why this is not redundant with `shadowingSelf`.

Reply:
The difference is that `shadowingSelf` dissects the result of an EVAL, not a field that is loaded straight from an index, so the analyzer is not impacted.
I'll add a comment to clarify the scope of the second test (and I'll link the issue) 👍